qa/tasks/keystone.py

```diff
@@ -417,6 +422,22 @@ def assign_ports(ctx, config, initial_port):
 
     return role_endpoints
 
+def read_ec2_credentials(ctx, client, user):
+    """
+    Look up EC2 credentials for the given user.
+    """
+    public_host, public_port = ctx.keystone.public_endpoints[client]
+    proc = run_in_keystone_venv(ctx, client,
+        ['openstack', 'ec2', 'credentials', 'list',
+         '--user', user, '--format', 'json', '--debug'] +
+        os_auth_args(public_host, public_port),
+        stdout=StringIO())
+    response = json.loads(proc.stdout.getvalue())
```

Comment:
making some progress:
```
2023-08-08T16:12:06.337 INFO:teuthology.orchestra.run.smithi062.stdout:[
2023-08-08T16:12:06.337 INFO:teuthology.orchestra.run.smithi062.stdout:  {
2023-08-08T16:12:06.337 INFO:teuthology.orchestra.run.smithi062.stdout:    "Access": "6351f6d729454264ba27a4a1832c3f8d",
2023-08-08T16:12:06.337 INFO:teuthology.orchestra.run.smithi062.stdout:    "Secret": "651d8ede06934828bb5666b996d22410",
2023-08-08T16:12:06.337 INFO:teuthology.orchestra.run.smithi062.stdout:    "Project ID": "5ce5ea6c00ae47fa9edb966b12b27686",
2023-08-08T16:12:06.338 INFO:teuthology.orchestra.run.smithi062.stdout:    "User ID": "5c692f3ae22a417bac41acab291d81e1"
2023-08-08T16:12:06.338 INFO:teuthology.orchestra.run.smithi062.stdout:  }
2023-08-08T16:12:06.338 INFO:teuthology.orchestra.run.smithi062.stdout:]
2023-08-08T16:12:06.469 ERROR:teuthology.contextutil:Saw exception from nested tasks
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_teuthology_7fda95956ac10132c9b74016ba832db907df09fa/teuthology/contextutil.py", line 30, in nested
    vars.append(enter())
  File "/usr/lib/python3.8/contextlib.py", line 113, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/github.com_cbodley_ceph_91cc18c91d8eafc57e15922762084e7051d083a6/qa/tasks/s3tests.py", line 114, in create_users
    access, secret = ctx.keystone.read_ec2_credentials(ctx, **keystone_user)
  File "/home/teuthworker/src/github.com_cbodley_ceph_91cc18c91d8eafc57e15922762084e7051d083a6/qa/tasks/keystone.py", line 435, in read_ec2_credentials
    response = json.loads(proc.stdout.getvalue())
AttributeError: 'NoneType' object has no attribute 'stdout'
```

Reply:
not sure what it's complaining about now
```
    response = json.loads(proc.stdout.getvalue())
AttributeError: 'list' object has no attribute 'stdout'
```
from http://qa-proxy.ceph.com/teuthology/cbodley-2023-08-08_18:09:50-rgw:tempest-main-distro-default-smithi/7363371/teuthology.log

Reply:
Usually, that means the command (credentials list?) didn't have any output.  Bad connection or crash, maybe?

Reply:
got this part working:
```
2023-08-09T16:54:20.630 INFO:tasks.keystone:proc: [RemoteProcess(client=<paramiko.client.SSHClient object at 0x7f931a84b970>, args=['cd', '/home/ubuntu/cephtest/keystone', Raw('&&'), 'source', '.tox/venv/bin/activate', Raw('&&'), 'openstack', 'ec2', 'credentials', 'list', '--user', 's3tests-main', '--format', 'json', '--debug', '--os-username', 'admin', '--os-password', 'ADMIN', '--os-user-domain-id', 'default', '--os-project-name', 'admin', '--os-project-domain-id', 'default', '--os-identity-api-version', '3', '--os-auth-url', 'http://smithi177.front.sepia.ceph.com:5000/v3'], check_status=True, hostname='smithi177')]
2023-08-09T16:54:20.630 INFO:tasks.keystone:response: [{'Access': 'f0fdcaa1f1ac4b58ba6c46ab10885b49', 'Secret': '88f51ec220db44499c0dd6145342f7b9', 'Project ID': 'cf23225fc9af4c54a8a310dec694b53a', 'User ID': '4ee9cbac66ac45fc862a11a7adb5683b'}]
2023-08-09T16:54:20.630 DEBUG:tasks.s3tests:Using keystone user s3tests-main credentials (f0fdcaa1f1ac4b58ba6c46ab10885b49 : 88f51ec220db44499c0dd6145342f7b9) for foo.client.0 on client.0
2023-08-09T16:54:20.631 DEBUG:tasks.s3tests:Creating user bar.client.0 on client.0
```